### DISCUSSION TITLE
Validation set for the Auto models in MLForecast

### DISCUSSION THREAD

**Discussion by blackhole-canada** [Nixtla/mlforecast#467] [Q&A] [answered]
Can I use a validation set for hyperparameter tuning in MLForecast, similar to NeuralForecast? Specifically, I want to use 12 months as a validation set to test on another 12 months. If I set the window to 1 in AutoML.fit() and then make predictions for the test period, does that effectively achieve this? I'm unsure because I don't see anything about a validation set mentioned in the documentation. Thank you in advance :)

**Comment by jmoralez** [comment, marked as answer]
Hey. Yes, setting `n_windows=1` uses a single validation set to choose the best parameters and then a final model is trained on all data, so predicting afterwards produces forecasts for the future, which I guess matches your test set.
